### PR TITLE
Rebrand sml-sync to faculty-sync

### PR DESCRIPTION
**Pull request by tcbegley**
ASI Data Science (formally "Advanced Skills Initiative Limited", UK company number 08873131) has been renamed "Faculty Science Limited" as of 4th February 2019.

This PR removes all references to ASI Data Science and SherlockML, updating them to Faculty. It also updates some of the implementation that relied on `sml` to `faculty` and `faculty_cli`.

I haven't updated the gif in the readme which still has SherlockML branding in it.

@sbalian @pbugnion

### CODE REVIEW

**Comment by pbugnion** [comment]
Thanks so much for doing this. I'll take a look tomorrow morning.

**Comment by pbugnion** [comment]
I released an alpha version off `d14f562`. You should now be able to:

```
pip install faculty-sync
```

**Comment by tcbegley** [comment]
Good call on the configuration files.

We used `faculty-cli` initially to minimise changes, but your reasons for wanting to ditch it make sense. I'm happy to try that if you don't mind? It's a good excuse for me to get a bit more familiar with the `faculty` library.

**Comment by tcbegley** [comment]
@pbugnion removing `faculty-cli` turned out to be pretty straightforward, so this is ready for another look.

`faculty_cli.shell` is just vendoring `shlex.quote` for Python 2 compatibility, but since `faculty-sync` is already Python 3 only I just imported `shlex.quote` directly instead. `faculty_cli.auth` I have replaced with the accounts client from `faculty`.

File: README.md [hunk 1]
```diff
@@ -1,14 +1,14 @@
-SherlockML incremental synchronization
+Faculty Platform incremental synchronization
 ======================================
```
Comment (pbugnion) [inline README.md]: The underline length should match the title

Comment (tcbegley) [reply README.md]: Thanks, will amend.

**Comment by pbugnion** [comment]
This is great, thank you!

**Pull request merged**